src/Xml/Dom/Configurator/pretty_print.php

```diff
@@ -5,15 +5,16 @@
 namespace VeeWee\Xml\Dom\Configurator;
 
 use Closure;
-use DOMDocument;
+use \DOM\XMLDocument;
 
 /**
- * @return Closure(DOMDocument): DOMDocument
+ * @return Closure(\DOM\XMLDocument): \DOM\XMLDocument
  */
 function pretty_print(): Closure
 {
-    return static function (DOMDocument $document): DOMDocument {
-        $document->preserveWhiteSpace = false;
+    return static function (\DOM\XMLDocument $document): \DOM\XMLDocument {
+        // TODO :  not fully implemented yet in the new API
+        //$document->preserveWhiteSpace = false;
```

Comment:
TODO : preserveWhiteSpace does not seem to be implemented and there is no way to specify from constructor wether to trim or pretty print.

Reply:
Actually, I completely missed that the necessary parse option is already exposed, sorry.
Passing `LIBXML_NOBLANKS` to the parse options will have the exact same effect as setting `preserveWhiteSpace` to `false`.

Reply:
That works. Thanks